api/controllers/user.js

```diff
@@ -319,12 +342,59 @@ function loginUser(req, res) {
         birthdate: moment(user.birthdate).format('YYYY-MM-DD'),
         authToken: tokenString
       };
-
       return res.status(200).json(response);
     }
   });
 }
 
+async function updateUserLocation(ip, user) {
+  if (!user.location || !user.location.country) {
+    const newLocation = await getLocation(ip);
+    if (newLocation && newLocation.country) {
+      user.location = newLocation;
+      try {
+        const dbUser = await user.save();
+        if (!dbUser) {
+          user.location = {};
```

Comment:
i think that is better to set `user.location = null`
for me, it is more clear that the location is not set